di2f/index.md

```diff
@@ -0,0 +1,226 @@
+      "serviceProtocol": "/thread/0.0.1"
+    }
+  ]
+}
+```
+
+</div>
+
+# Method Definition
+
+The crux of any DID implementation is defining it's _DID method_. The DID specification defines a _method_ as a "means to implement this specification on different verifiable data registries" [@reed2020decentralized (sec 8)]. In other words, new DID methods should be specified such that different implementations of the same DID method remain interoperable.
+
+<!-- @sanderpick Maybe that last statement is "weak", drop it? Something better to put there? -->
+
+Here we define a DID method specification for threads which is composed of a _method scheme_ (see [@reed2020decentralized] sec 3.1) and _operations_ (sec 8.2). Operations specify how a DID document is created, how to read/verify a document, as well as how a DID controller can update or even deactivate a DID document. The method scheme defines the structure of the DID implementation's string identifier(s).
+
+## Method Scheme
+
+A thread DID method scheme prefixes the unique identifier for a thread (see [@pick2020protocol] sec 2.2), with the globally unique `did:thread:` namespace, such that a thread DID becomes `did:thread:<thread-id>`, where `thread-id` is defined as:
+
+$$ \underbrace{\texttt{0x62}}_\text{Multibase} \overbrace{\texttt{0x01}}^\text{Version} \underbrace{\texttt{0x55}}_\text{Variant} \overbrace{\texttt{0x539bc}\dots\texttt{a4546e}}^\text{Random Component} $$
+
+This produces a string identifier of the form: `"did:thread:bafk6npbyp...6mfuhoe6iesr"`
+
+## Method Operations
+
+As mention perviously, DID methods define a set of operations that can be performed on/with DID documents. DID implementations often use a smart contract [@buterin2014next] on a blockchain like Ethereum to model the global data registry, and to implement these operations. For the implementation proposed here, we assume that documents are stored on the Filecoin blockchain as non-fungible tokens (NFTs). Filecoin does not yet support NFTs, but there is an informal proposal to create a new _actor_ type for them.
+
+<!-- @sanderpick this is a pretty "weak" statement :( While totally sufficient for a github discussion, I don't think it'll fly here. What else might we be able to say here instead? Do we have anything we can reference? -->
+
+In our initial proof-of-concept network, a non-consensus driven global data registry is used, based on a p2p "gossip" protocol. In practice, this is implemented using libp2p's gossip-sub implementation. This registry provides weak consensus, along the lines of IPNS-over-pubsub [@santos2020IPNS].
+
+The process of getting a DID subject's document from the verifiable data registry is called _resolution_. Any peer can resolve any document by querying the on-chain NFT representing the subject, or in the shoter-term case, by posting queries to the associated pubsub channel.
+
+# Architecture
+
+<!-- @sanderpick What is the take home point of this section? -->
```

Comment:
@sanderpick What is the take home point of this section?

Reply:
I guess the "take home" would be "here are a few possible ways to offer services advertised via a thread DID"